### DISCUSSION TITLE
Board Permissions

### DISCUSSION THREAD

**Discussion by JStuve** [microsoft/vsts-extension-retrospectives#635] [Ideas]
Looking for input on a well desired feature for board-based permissions. Below is a list of features, mocks, existing issues, discussions and contributors. The goal of this discussion is to consolidate all existing feedback items and receive feedback from the proposed solution outlined below.

### Summary

Permission-based boards to control who has access to the Retrospective, improving the "safe environment" across teams. Allowing users to assign team member permissions at the board level, giving the highest flexibility on who can participate in the retrospective.

### Feature List
* When creating a board a new tab will allow you to manage who has access to the board (Creating a copy will also persist the team members).
* Boards will only be visible, and accessible, if the team member has access.
* Checkbox setting, which will be enabled by default, to allow everyone to view the board. Unchecking the box will automatically add every team member to the board.
* Search bar will allow to bulk add teams and/or members as needed.
* Table checkbox will allow you bulk remove team members as needed.

![azure-retro-permission-poc-1](https://github.com/microsoft/vsts-extension-retrospectives/assets/14914689/75ad6b3d-1617-4b94-8fe3-5bd4db09315a)

_Design choices from the mocks may not be a 1 to 1 come release._

### Related Issues/Discussion
* #309
* #474
* #498
* #389 
* #342
* #212 
* #145
* #49
* #5

### Contributes
List of users who show interest in this feature. I want your feedback!

* @dberkovich01
* @Enterprise-Agile-Coach-C
* @tthb1
* @rschue
* @justdait

**Comment by polatengin** [comment]
I like the idea of having a `Permissions` tab in the `Create new Retrospective` dialog 👍

Is it possible to accept teams, too? Like, `Search users/teams` so if a team is already setup for a project, we can re-use the team.

**Reply by JStuve** [reply]
Definitely, will add in teams.

**Comment by dphanson63** [comment]
I see a Permissions tab on my Retrospective boards, but it appears to be very buggy, and not functional.  Is this feature supposed to be active in ADO at this time?

**Comment by dphanson63** [comment]
@JStuve Looks like this feature is working now.  Is it time to close the discussion?
